libs/vm_modules/include/vm_modules/core/system.hpp

```diff
@@ -0,0 +1,122 @@
+#pragma once
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "core/commandline/parameter_parser.hpp"
+#include "vm/module.hpp"
+
+namespace fetch {
+namespace vm_modules {
+
+class Parameters
+{
+public:
+  using ParamParser = fetch::commandline::ParamsParser;
+  using ArgList     = std::vector<char *>;
+  using StringList  = std::vector<std::string>;
+
+  void Parse(int argc, char **argv)
+  {
```

Comment:
Additionally, I think we could move all the function definitions in this file into a cpp file.

Reply:
Done!